st2client/st2client/commands/resource.py

```diff
@@ -314,9 +316,15 @@ def run(self, args, **kwargs):
         return self.manager.create(instance, **kwargs)
 
     def run_and_print(self, args, **kwargs):
-        instance = self.run(args, **kwargs)
-        self.print_output(instance, table.PropertyValueTable,
-                          attributes=['all'], json=args.json)
+        try:
+            instance = self.run(args, **kwargs)
+            if not instance:
+                raise Exception('Server did not create instance.')
+            self.print_output(instance, table.PropertyValueTable,
+                              attributes=['all'], json=args.json)
+        except Exception as e:
+            print('ERROR: %s' % e.message)
+            raise OperationFailureException('Create failed.')
```

Comment:
Maybe we should also include original error message in the derived exception message.